admin_setting_oidc_integration_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfe
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type Jwks struct {
+	Keys []Key `json:"keys"`
+}
+
+type Key struct {
+	Kid string `json:"kid"`
+}
+
```

Comment:
Since this is the main tfe package, I would recommend renaming these to something a little more specific (and making them private for good measure)

```suggestion
type wellKnownJwks struct {
	Keys []struct {
		Kid string `json:"kid"`
	} `json:"keys"`
}

```